### ISSUE TITLE
Fatal error - TypeError CompatModule.php

### ISSUE THREAD

**Issue by jurica-zecevic** [woocommerce/woocommerce-paypal-payments#2416]
### Describe the Bug

Fatal error
: Uncaught TypeError: WooCommerce\PayPalCommerce\Compat\CompatModule::WooCommerce\PayPalCommerce\Compat\{closure}(): Argument #3 ($file) must be of type string, null given, called in /home/site-name/public_html/wp-includes/class-wp-hook.php on line 324 and defined in /home/site-name/public_html/wp-content/plugins/woocommerce-paypal-payments/modules/ppcp-compat/src/CompatModule.php:371 Stack trace: #0 /home/site-name/public_html/wp-includes/class-wp-hook.php(324): WooCommerce\PayPalCommerce\Compat\CompatModule->WooCommerce\PayPalCommerce\Compat\{closure}(false, '<script id="coo...', NULL) #1 /home/site-name/public_html/wp-includes/plugin.php(205): WP_Hook->apply_filters(false, Array) #2 /home/site-name/public_html/wp-content/plugins/w3-total-cache/Minify_AutoJs.php(214): apply_filters('w3tc_minify_js_...', false, '<script id="coo...', NULL) #3 /home/site-name/public_html/wp-content/plugins/w3-total-cache/Minify_AutoJs.php(124): W3TC\Minify_AutoJs->process_script_tag('<script id="coo...', 0) #4 /home/site-name/public_html/wp-content/plugins/w3-total-cache/Minify_Plugin.php(202): W3TC\Minify_AutoJs->execute() #5 [internal function]: W3TC\Minify_Plugin->ob_callback('<!doctype html>...') #6 /home/site-name/public_html/wp-content/plugins/w3-total-cache/Util_Bus.php(21): call_user_func(Array, '<!doctype html>...') #7 /home/site-name/public_html/wp-content/plugins/w3-total-cache/Generic_Plugin.php(537): W3TC\Util_Bus::do_ob_callbacks(Array, '<!doctype html>...') #8 [internal function]: W3TC\Generic_Plugin->ob_callback('<!doctype html>...', 9) #9 /home/site-name/public_html/wp-includes/functions.php(5420): ob_end_flush() #10 /home/site-name/public_html/wp-includes/class-wp-hook.php(324): wp_ob_end_flush_all('') #11 /home/site-name/public_html/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array) #12 /home/site-name/public_html/wp-includes/plugin.php(517): WP_Hook->do_action(Array) #13 /home/site-name/public_html/wp-includes/load.php(1270): do_action('shutdown') #14 [internal function]: shutdown_action_hook() #15 {main} thrown in
/home/site-name/public_html/wp-content/plugins/woocommerce-paypal-payments/modules/ppcp-compat/src/CompatModule.php
on line
371

- No error on plugin version 2.8.0

### Environment

 - WordPress Version  6.5.5
 - WooCommerce Version 9.0.2
 - Plugin Version 2.8.1
 - Browser [e.g. Chrome, Safari] and Version - Chrome Version 126.0.6478.127 (Official Build) (64-bit)
 - Any other plugins installed

<details><summary>System status</summary>

  ```
  <details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.0.2
Legacy REST API Package Version: Legacy REST API dodatak nije instaliran na ovoj web-stranici.
Action Scheduler Version: ✔ 3.7.4
Log Directory Writable: ✔
WP Version: 6.5.5
WP Multisite: –
WP Memory Limit: 1 GB
WP Debug Mode: –
WP Cron: ✔
Language: hr
External object cache: ✔

### Server Environment ###

Server Info: Apache
PHP Version: 8.1.29
PHP Post Max Size: 50 MB
PHP Time Limit: 180
PHP Max Input Vars: 6000
cURL Version: 7.87.0
OpenSSL/1.1.1w

SUHOSIN Installed: –
MySQL Version: 10.6.18-MariaDB-cll-lve-log
Max Upload Size: 50 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

a2w_shipping: 20
attachment: 11140
customize_changeset: 1
gift_card: 1
nav_menu_item: 44
nectar_slider: 7
oembed_cache: 254
order_status: 1
page: 14
post: 14
product: 484
product_variation: 790
revision: 1210
salient_g_sections: 1
shop_coupon: 3
shop_order: 61
wp_global_styles: 1
wpcf7_contact_form: 1
yith_wcan_preset: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (35) ###

Akismet Anti-spam: Spam Protection: od Automattic - Anti-spam Team – 5.3.3
Ali2Woo: od Ali2Woo Team – 3.2.4
Autocomplete Address and Location Picker for WooCommerce: od powerfulwp – 1.1.8
Custom Order Status Manager for WooCommerce: od Bright Plugins – 1.1.3.2
Contact Form 7: od Takayuki Miyoshi – 5.9.6
CookieYes | GDPR pristanak na kolačiće: od CookieYes – 3.2.4
Disabled Source, Disabled Right Click and Content Protection: od jahidcse – 1.4.1
Drag and Drop Multiple File Upload - Contact Form 7: od Glen Don L. Mongaya – 1.3.8.1
GTranslate: od Translate AI Multilingual Solutions – 3.0.6
OMGF: od Daan from Daan.dev – 5.9.0
Salient WPBakery Page Builder: od Michael M - WPBakery.com | Modified by ThemeNectar – 7.7.2
Loco Translate: od Tim Whitlock – 2.6.10
oneTap: od Exlac – 1.0.9
Salient Core: od ThemeNectar – 2.1.0
Salient Nectar Slider: od ThemeNectar – 1.7.6
Salient Shortcodes: od ThemeNectar – 1.5.4
Salient Social: od ThemeNectar – 1.2.5
Salient Widgets: od ThemeNectar – 1.3
Super Progressive Web Apps: od SuperPWA – 2.2.28
SVG Support: od Benbodhi – 2.5.5
W3 Total Cache: od BoldGrid – 2.7.3
CTX Feed: od WebAppick – 6.5.0
MailerLite - WooCommerce integration: od MailerLite – 2.1.16
WooCommerce Stripe Gateway: od WooCommerce – 8.5.0
WooCommerce Najniža cijena: od Igor Kovacic – 1.0.2
WooCommerce MX R1 račun: od Media X – 1.5
PayCek payment: od Leftor d.o.o – 1.0.7.
WooCommerce PayPal Payments: od WooCommerce – 2.8.1
WooCommerce: od Automattic – 9.0.2 (dostupno je ažuriranje na verziju 9.1.1)
Yoast SEO: od Tim Yoast – 23.0
WPC Estimated Delivery Date for WooCommerce: od WPClever – 2.4.3
YITH WooCommerce Ajax Product Filter: od YITH – 5.3.0
YITH WooCommerce Brands Add-On: od YITH – 2.25.0
YITH WooCommerce Gift Cards: od YITH – 4.14.0
YITH WooCommerce Waitlist: od YITH – 2.16.0

### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php
db.php: db.php
maintenance.php: maintenance.php
object-cache.php: object-cache.php

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: right_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
gift-card (gift-card)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: –
Order datastore: WC_Order_Data_Store_CPT
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 dana
Level threshold: –
Log directory size: 230 KB

### WC Pages ###

Osnova trgovine: #10 - /shop/
Košarica: #11 - /kosarica/
Blagajna: #12 - /blagajna/
Moj račun: #13 - /moj-racun/
Odredbe i uvjeti: #5901 - /uvjeti-pruzanja-usluge/

### Theme ###

Name: Salient Child Theme
Version: 1.0
Author URL: https://themeforest.net/user/themenectar
Child Theme: ✔
Parent Theme Name: Salient
Parent Theme Version: 16.3.0
Parent Theme Author URL: https://themeforest.net/user/themenectar
WooCommerce Support: ✔

### Templates ###

Overrides: salient-child/woocommerce/cart/cross-sells.php
salient/woocommerce/cart/mini-cart.php
salient/woocommerce/checkout/form-checkout.php
salient/woocommerce/checkout/review-order.php
salient/woocommerce/checkout/terms.php
salient/woocommerce/content-product.php
salient/woocommerce/content-single-product.php
salient/woocommerce/loop/add-to-cart.php
salient/woocommerce/loop/loop-start.php
salient/woocommerce/myaccount/form-login.php
salient/woocommerce/single-product/product-image.php
salient/woocommerce/single-product/rating.php
salient/woocommerce/single-product/tabs/description.php
salient/woocommerce/single-product/tabs/tabs.php
salient/woocommerce/single-product/title.php
salient-child/woocommerce/single-product/up-sells.php
salient/woocommerce/single-product-reviews.php

### WooCommerce PayPal Payments ###

Onboarded: ✔
Shop country code: HR
WooCommerce currency supported: ✔
Advanced Card Processing available in country: –
Pay Later messaging available in country: –
Webhook status: –
PayPal Vault enabled: –
ACDC Vault enabled: –
Logging enabled: –
Reference Transactions: –
Used PayPal Checkout plugin: –
Subscriptions Mode: Disabled

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
pattern-toolkit-full-composability
product-pre-publish-modal
settings
async-product-editor-category-field
launch-your-store
product-editor-template-system

Daily Cron: ✔ Next scheduled: 2024-07-13 12:40:05 +02:00
Options: ✔
Notes: 111
Onboarding: completed

### Action Scheduler ###

Dovrši: 1.509
Oldest: 2024-06-12 06:27:41 +0200
Newest: 2024-07-12 15:45:55 +0200

Neuspjelo: 413
Oldest: 2023-09-08 10:40:46 +0200
Newest: 2024-06-19 14:08:52 +0200

U pripremi: 1
Oldest: 2024-07-13 11:34:05 +0200
Newest: 2024-07-13 11:34:05 +0200

### Status report information ###

Generated at: 2024-07-12 16:07:45 +02:00
```
</details>
  ```

</details>

**Comment by InpsydeNiklas** [comment]
Hi @jurica-zecevic, thanks for reporting this error. The problem was resolved for the next 2.8.2 update, which can be tested from here: https://github.com/woocommerce/woocommerce-paypal-payments/releases/tag/2.8.2-rc1
The official release is planned for the coming week. I hope this helps!

**Comment by jurica-zecevic** [comment]
Hi @InpsydeNiklas Thank you :)
